src/code/z_camera.c

```diff
@@ -543,323 +866,7417 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
+    f32 sp50;
+    PosRot* focalActorPosRot = &camera->focalActorPosRot;
+
+    focalActorAtOffsetTarget.y = focalActorHeight + yOffset;
+    focalActorAtOffsetTarget.x = 0.0f;
+    focalActorAtOffsetTarget.z = 0.0f;
+
+    if (PREG(76) && (flags & FLG_ADJSLOPE)) {
+        focalActorAtOffsetTarget.y -=
+            Camera_CalcSlopeYAdj(&camera->floorNorm, camera->focalActorPosRot.rot.y, eyeAtDir->yaw, 25.0f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Special7.s")
+    atTarget = focalActorPosRot->pos;
+    atTarget.y += focalActorHeight;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Special8.s")
+    OLib_Vec3fDiffToVecGeo(outPlayerToTargetDir, &atTarget, targetPos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Special9.s")
+    sp5C = *outPlayerToTargetDir;
+    sp5C.r = (distance < sp5C.r) ? (sp5C.r * 0.2f) : ((sp5C.r * 0.9f) - (sp5C.r * 0.7f * (sp5C.r / distance)));
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Alloc.s")
+    if (flags & FLG_OFFGROUND) {
+        sp5C.r *= 0.2f;
+        camera->yOffsetUpdateRate = camera->xzOffsetUpdateRate = 0.01f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Free.s")
+    OLib_VecGeoToVec3f(&sp68, &sp5C);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Init.s")
+    focalActorAtOffsetTarget.x += sp68.x;
+    focalActorAtOffsetTarget.y += sp68.y;
+    focalActorAtOffsetTarget.z += sp68.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800DDFE0.s")
+    if (func_800CB950(camera)) {
+        *yPosOffset = Camera_ScaledStepToCeilF(focalActorPosRot->pos.y, *yPosOffset, 0.4f, 0.1f);
+        deltaY = focalActorPosRot->pos.y - *yPosOffset;
+        focalActorAtOffsetTarget.y -= deltaY;
+        Camera_ScaledStepToCeilVec3f(&focalActorAtOffsetTarget, &camera->focalActorAtOffset, camera->xzOffsetUpdateRate,
+                                     camera->yOffsetUpdateRate, 0.1f);
+    } else {
+        if (!(flags & 0x80)) {
+            deltaY = focalActorPosRot->pos.y - *yPosOffset;
+            sp50 = OLib_Vec3fDistXZ(at, &camera->eye);
+            phi_f16 = sp50;
+            func_80086B30(deltaY, sp50);
+            fovHeight = func_80086760(DEG_TO_RAD(camera->fov * 0.4f)) * phi_f16;
+
+            if (fovHeight < deltaY) {
+                *yPosOffset += deltaY - fovHeight;
+                deltaY = fovHeight;
+            } else if (deltaY < -fovHeight) {
+                *yPosOffset += deltaY + fovHeight;
+                deltaY = -fovHeight;
+            }
+            focalActorAtOffsetTarget.y -= deltaY;
+        } else {
+            deltaY = focalActorPosRot->pos.y - *yPosOffset;
+            temp_f0_6 = func_80086B30(deltaY, OLib_Vec3fDistXZ(at, &camera->eye));
+
+            if (temp_f0_6 > 0.34906584f) { // (M_PI / 9)
+                phi_f16 = 1.0f - sin_rad(temp_f0_6 - 0.34906584f);
+            } else if (temp_f0_6 < -0.17453292f) { // (M_PI / 18)
```

Comment:
It does not no, although more is known about floats now then when I tried it, so maybe there's a trick I'm not aware of that can help

Reply:
I see, my only suggestion would be to update them to something like `// close to (M_PI / 18)` or something like that. The point being to highlight that it does *not* match if you use that fraction, but that if you're trying to reason about the math, the comment tells you what our best guess is for this.

Reply:
It's only 1 bit off though, so it's negligibly different from `(M_PI / 18)`, so that's definitely the true value. It's probably something like `(2 * M_PI / 36)` or `(10 * M_PI / 180)`or something similar. Not sure how to phrase that as a comment

Reply:
Ah, I didn't realize it was so close. In that case, it's fine to leave it as-is IMO.